packages/components/src/navigation/components-composition-navigation.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState, useRef } from '@wordpress/element';
+import { Icon, chevronLeft, chevronRight } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Animate from '../animate';
+import {
+	BackButtonUI,
+	MenuItemTitleUI,
+	MenuItemUI,
+	MenuTitleUI,
+	MenuUI,
+	Root,
+} from './styles/navigation-styles';
+import Button from '../button';
+
+export default function ComponentsCompositionNavigation( {
+	initialActiveLevel,
+	children,
+} ) {
+	const [ activeLevel, setActiveLevel ] = useState( initialActiveLevel );
+	const [ slideOrigin, setSlideOrigin ] = useState( 'left' );
+
+	const isMounted = useRef( false );
+	useEffect( () => {
+		if ( ! isMounted.current ) {
+			isMounted.current = true;
+		}
+	}, [] );
+
+	const navigateTo = ( level ) => {
+		setActiveLevel( level );
+		setSlideOrigin( 'left' );
+	};
+
+	const navigateBack = ( level ) => {
+		setActiveLevel( level );
+		setSlideOrigin( 'right' );
+	};
+
+	function NavigationLevel( {
+		children: levelChildren,
+		slug,
+		title,
+		parentLevel,
+		parentTite,
+	} ) {
+		if ( activeLevel !== slug ) {
+			return null;
+		}
+
+		return (
+			<div className="components-navigation__level">
+				{ parentLevel ? (
+					<BackButtonUI
```

Comment:
Nice!  If we can extract this to its own component with an `onClick` prop then we can resolve the back button in the wc nav repo.